### ISSUE TITLE
Right-Click With Folder Selected + New File

### ISSUE THREAD

**Issue by ghost** [atom/tree-view#252]
When you select a directory, then right-click below your file tree (not on a file), then click "New File", it creates the file starting with your selected directory instead of the root. This may be meant to be a feature, but for me this feels counter-intuitive.

![rightclick](https://cloud.githubusercontent.com/assets/4193418/4982795/7c1567f2-6915-11e4-85d7-58b997dec3ee.gif)

**Comment by Dayjo** [comment]
For me, this is expected functionality. Perhaps it's more about the fact that you right clicked away from that directory. 

For instance, if you right click on the specific directory I would expect the above functionality, but if you right click on no directory, then yes, it should create it in the root.

**Comment by GymbylCoding** [comment]
Yeah; that's my point. It should only create the file in that directory if you right click *on* the directory. Below the directory should mean root.

**Comment by Dayjo** [comment]
Ah yes sorry, didn't read properly :)

I'd also like a keyboard shortcut for this new file command on the current selected directory. There's a couple of packages that do 'advanced new/open file' which tend to only base on the directory of the active file. For me it'd be great to be able to set that to base the directory on the current selected directory in the files list  :+1:

**Comment by bgriffith** [comment]
Just adding my agreement that it feels counter intuitive. For some reason I rarely remember to clear the path and end up wondering where my files/folders have got to :smile:  Was going to post until I discovered this thread :+1:

**Comment by UziTech** [comment]
Not sure if this should be in this thread (kind of the opposite of this problem)

When selecting multiple entries then right clicking on those entries only the right clicked entry is left selected.

![right-click-tree-view2](https://cloud.githubusercontent.com/assets/97994/26324243/9e885078-3ef7-11e7-8fa2-6163a8078235.gif)

I don't think this used to be the case. All selected entries should be left selected.

**Comment by UziTech** [comment]
I found the bug for my issue in [/lib/tree-view.coffee#L806](https://github.com/atom/tree-view/blob/8b89251c5789d1486242ef4348f6f5c5196c6059/lib/tree-view.coffee#L806)

PR submitted #1115 

```diff
 ...
      # return early if we're opening a contextual menu (right click) during multi-select mode
      if @multiSelectEnabled() and
-        e.target.classList.contains('selected') and
+        entryToSelect.classList.contains('selected') and
         # mouse right click or ctrl click as right click on darwin platforms
         (e.button is 2 or e.ctrlKey and process.platform is 'darwin')
        return
 ...
```

it seems like there could be an easy fix for your issue


```diff
  onMouseDown: (e) ->
    if entryToSelect = e.target.closest('.entry')
  ...
+   else
+     @deselect()

```

right now if you don't click on an entry it just does nothing (i.e. leaves the selected entries selected)

or if you want is to only deselect on right click


```diff
  onMouseDown: (e) ->
    if entryToSelect = e.target.closest('.entry')
  ...
+   else if e.button is 2 or e.ctrlKey and process.platform is 'darwin'
+     @deselect()

```
